### PR TITLE
Fix some memory leak issues in libr/core/cbin.c flagged by coverity

### PR DESCRIPTION
**Pull request by MewtR**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/core/cbin.c [hunk 1]
```diff
@@ -3383,6 +3387,9 @@ static bool bin_sections(RCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
 	free (hashtypes);
 	r_table_free (table);
 	ht_pp_free (dup_chk_ht);
+	if (io_section_info) {
```
Comment (trufae) [inline libr/core/cbin.c:3390]:
```suggestion
```

File: libr/core/cbin.c [hunk 2]
```diff
@@ -3383,6 +3387,9 @@ static bool bin_sections(RCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
...
+		r_list_free (io_section_info);
```
Comment (trufae) [inline libr/core/cbin.c:3391]:
```suggestion
	r_list_free (io_section_info);
```

File: libr/core/cbin.c [hunk 3]
```diff
@@ -3383,6 +3387,9 @@ static bool bin_sections(RCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
...
+	}
```
Comment (trufae) [inline libr/core/cbin.c:3392]:
```suggestion
```

**Comment by trufae** [comment]
run r2r -i db/formats/dwarf

**Comment by MewtR** [comment]
Wow, that's really cool. Didn't know interactive mode allowed you to fix tests like that.

**Pull request merged**